hydra-node/src/Hydra/Network/BroadcastToSelf.hs

```diff
@@ -0,0 +1,22 @@
+module Hydra.Network.BroadcastToSelf where
+
+import Cardano.Prelude
+import Control.Monad.Class.MonadThrow (MonadThrow)
+import Hydra.Network (Network (..), NetworkCallback, NetworkComponent)
+
+-- | Ensures that messages `broadcast` through the wrapped `NetworkComponent` are
+-- also sent back to "self" through passed `NetworkCallback`.
+withBroadcastToSelf ::
```

Comment:
I find the name a little bit confusing, for there's an emphasis on 'self' but it's actually broadcasting (i.e. to everyone) AND sending to self. I don't have a better name to suggest though..

Reply:
Yes, the somewhat contradictory nature of the name sounded like a nice pun to me but better avoid pun in code. Feel free to rename once you find a better one.